zio-http/src/main/scala/zio/http/netty/NettyBodyWriter.scala

```diff
@@ -66,27 +68,45 @@ object NettyBodyWriter {
         None
       case StreamBody(stream, _, _)           =>
         Some(
-          stream.chunks
-            .runFoldZIO(Option.empty[Chunk[Byte]]) {
-              case (Some(previous), current) =>
+          contentLength match {
+            case Some(length) =>
+              stream.chunks
+                .runFoldZIO(length) { (remaining, bytes) =>
+                  remaining - bytes.size match {
+                    case 0L =>
+                      NettyFutureExecutor.executed {
+                        // Flushes the last body content and LastHttpContent together to avoid race conditions.
+                        ctx.write(new DefaultHttpContent(Unpooled.wrappedBuffer(bytes.toArray)))
+                        ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT)
```

Comment:
it is possible, if exceedingly unlikely, for an incorrect content-length header to be a multiple of the chunk size. this would result in writing `EMPTY_LAST_CONTENT` and then continuing to write more data as the `runFoldZIO` continues

Reply:
Yes it is. It's a kind of workaround as the size of the whole stream is indeterministic. That's why I didn't apply the same for the client side yet.